prog/postgres/postgres_server_nexus.rb

```diff
@@ -436,23 +445,25 @@ def before_run
   label def prepare_for_take_over
     decr_take_over
 
-    hop_taking_over if postgres_server.read_replica? || postgres_server.resource.representative_server.nil?
+    hop_read_replica_promote if postgres_server.read_replica?
+    hop_taking_over if postgres_server.resource.representative_server.nil?
 
     postgres_server.resource.representative_server.incr_destroy
 
     nap 5
   end
 
-  label def taking_over
-    if postgres_server.read_replica?
-      old_representative = postgres_server.resource.representative_server
-      old_representative.update(representative_at: nil)
-      postgres_server.update(representative_at: Time.now)
-      postgres_server.resource.incr_refresh_dns_record
-      old_representative.incr_destroy
-      hop_configure
-    end
+  label def read_replica_promote
```

Comment:
I'll think about how to reduce the branching and specialized code for read replicas vs. regular databases. Let's merge the parent PR first, though.

Reply:
The new version seems much cleaner. Thanks. :+1: